packages/store/addon/-private/system/weak-cache.ts

```diff
@@ -0,0 +1,70 @@
+import { DEBUG } from '@glimmer/env';
+
+import { addSymbol, symbol } from '../ts-interfaces/utils/symbol';
+
+interface WeakCache<K extends object, V> {
```

Comment:
Given that this class only provides value in debug, it seems unfortunate to ship the code in production.

Reply:
Discussed OOB (and since implemented) WeakCache directly extending WeakMap so as to reduce the footprint of what we need to do, but keeping the class since the extra sugar provides value even in production.